src/huggingface_hub/repocard.py

```diff
@@ -99,3 +104,162 @@ def metadata_eval_result(
         model_index, dict_factory=lambda x: {k: v for (k, v) in x if v is not None}
     )
     return {"model-index": [data]}
+
+
+def metadata_update(
+    repo_id: str,
+    metadata: Dict,
```

Comment:
```suggestion
    repo_id: str,
    *,
    metadata: Dict,
```

kwarg only to prevent API breakage is we need to re-order params later on? WDYT?

Reply:
`metadata` would need to have a default value in that case. May I offer the option of having `metadata` be before the `*`?
```suggestion
    repo_id: str,
    metadata: Dict,
    *
```

Reply:
I also thought they couldn't but it turns out they can have no default value:
https://stackoverflow.com/a/29037261

Reply:
yes it's just that they're then required (which is fine here IMO)